sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -67,9 +70,14 @@ class SimpleFunctionRegistry extends FunctionRegistry {
   }
 
   override def lookupFunction(name: String, children: Seq[Expression]): Expression = {
+    val builder = functionBuilders.get(name)
+    if (builder.isEmpty) {
+      throw new AnalysisException(s"undefined function $name")
+    }
     val func = synchronized {
-      functionBuilders.get(name).map(_._2).getOrElse {
-        throw new AnalysisException(s"undefined function $name")
+      Try(builder.map(_._2)) match {
```

Comment:
I still don't understand why this needs to be changed. This is just calling `Option.map` and then getting the second field, which is what the old code does as well.